common/whispers.ts

```diff
@@ -0,0 +1,73 @@
+export enum WhisperMessageType {
+  TextMessage = 'textMessage',
+}
+
+export interface BaseWhisperMessageData {
+  readonly type: WhisperMessageType
+}
+
+export interface WhisperTextMessageData extends BaseWhisperMessageData {
+  type: typeof WhisperMessageType.TextMessage
+  text: string
+}
+
+export type WhisperMessageData = WhisperTextMessageData
+
+export interface WhisperMessage {
+  id: string
+  from: WhisperUser
+  to: WhisperUser
+  sent: number
+  data: WhisperMessageData
+}
+
+// TODO(2Pac): Make this into an interface and include more information here
```

Comment:
not sure that's really desirable with where you're using this tbh, feels like this should just be an id in those places and any extra info about the users should be passed adjacently